### PR TITLE
Feature/semseg binary classification

### PR DESCRIPTION
**Pull request by dcaustin33**
Supports binary classification in dq for voxel

### CODE REVIEW

File: dataquality/integrations/torch_semantic_segmentation.py [hunk 1]
```diff
@@ -377,16 +388,26 @@ def get_argmax_probs(
         preds = F.interpolate(preds, size=input_shape, mode="bilinear")
 
         # checks whether the model is (n, classes, w, h), or (n, w, h, classes)
-        if preds.shape[1] == self.number_classes:
+        # takes the max in case of binary classification
+        if max(preds.shape[1], 2) == self.number_classes:
             preds = preds.permute(0, 2, 3, 1)
         assert (
-            preds.shape[-1] == self.number_classes
+            max(preds.shape[3], 2) == self.number_classes
```
Comment (setu4993) [inline dataquality/integrations/torch_semantic_segmentation.py:395]: We could still use the last index (`-1`), right?

File: dataquality/integrations/torch_semantic_segmentation.py [hunk 2]
```diff
@@ -377,16 +388,26 @@ def get_argmax_probs(
...
         logits = preds.clone()  # (bs, w, h, classes)
-        probs = (torch.nn.Softmax(dim=-1)(logits)).cpu()
+        if preds.shape[3] > 1:
+            probs = (torch.nn.Softmax(dim=-1)(logits)).cpu()
+        else:
+            probs = (torch.nn.Sigmoid()(logits)).cpu()
+
+        # expands the binary classification to a 2 channel tensor
+        if probs.shape[3] == 1:
+            probs = self.expand_binary_classification(probs)
```
Comment (setu4993) [inline dataquality/integrations/torch_semantic_segmentation.py:407]: We can move this above, right?
```suggestion
        else:
            probs = (torch.nn.Sigmoid()(logits)).cpu()
            # expands the binary classification to a 2 channel tensor
            probs = self.expand_binary_classification(probs)
```

**Review by setu4993 — COMMENTED** [review]
Useful update! Thanks, Derek!

If it's not too hard, would be nice to add a test for a dataset that does binary classification. Else, open to merging as-is.

File: dataquality/integrations/torch_semantic_segmentation.py (hunk 1 above)
Comment (dcaustin33) [reply dataquality/integrations/torch_semantic_segmentation.py:395]: We could either way I thought 3 might be more precise but either works - would love to merge so as to get it to voxel but we do need to add some semseg tests for sure

Comment (setu4993) [reply dataquality/integrations/torch_semantic_segmentation.py:395]: Sounds good, thanks!

**Review by setu4993 — APPROVED** [review]
@dcaustin33 : Can you please bump the version in `__init__.py` before merging so that we can release with a new version?

Thanks!

File: dataquality/integrations/torch_semantic_segmentation.py [hunk 3]
```diff
@@ -361,6 +361,17 @@ def calculate_mislabeled_pixels(
...
         return mislabeled_pixels
 
+    def expand_binary_classification(self, probs: torch.Tensor) -> torch.Tensor:
+        """Expands the binary classification to a 2 channel tensor
+
+        Args:
+            probs (torch.Tensor): binary classification tensor
+
+        Returns:
+            torch.Tensor: bs, 2, h, w tensor
```
Comment (anthonycorletti) [inline dataquality/integrations/torch_semantic_segmentation.py:371]: Is bs batch size?

**Review by anthonycorletti — APPROVED** [review]

File: dataquality/integrations/torch_semantic_segmentation.py (hunk 3 above)
Comment (dcaustin33) [reply dataquality/integrations/torch_semantic_segmentation.py:371]: Yep!

**Pull request merged**